android/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.kt

```diff
@@ -1,44 +1,45 @@
 package dev.openfeature.sdk
 
-import dev.openfeature.sdk.events.OpenFeatureEvents
-import dev.openfeature.sdk.events.awaitReadyOrError
-import dev.openfeature.sdk.events.observe
-import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.flow.Flow
+import dev.openfeature.sdk.exceptions.OpenFeatureError
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.SharedFlow
-import kotlinx.coroutines.flow.flatMapLatest
 
 @Suppress("TooManyFunctions")
 object OpenFeatureAPI {
     private val NOOP_PROVIDER = NoOpProvider()
     private var provider: FeatureProvider = NOOP_PROVIDER
     private var context: EvaluationContext? = null
-    private val providersFlow: MutableSharedFlow<FeatureProvider> = MutableSharedFlow(replay = 1)
-    internal val sharedProvidersFlow: SharedFlow<FeatureProvider> get() = providersFlow
+
+    private val _statusFlow: MutableSharedFlow<OpenFeatureStatus> =
+        MutableSharedFlow<OpenFeatureStatus>(replay = 1, extraBufferCapacity = 5)
+            .apply {
+                tryEmit(OpenFeatureStatus.NotReady)
+            }
+    val statusFlow: SharedFlow<OpenFeatureStatus> get() = _statusFlow
 
     var hooks: List<Hook<*>> = listOf()
         private set
 
-    fun setProvider(provider: FeatureProvider, initialContext: EvaluationContext? = null) {
+    suspend fun setProvider(provider: FeatureProvider, initialContext: EvaluationContext? = null) {
```

Comment:
This `suspend` is probably something we need to decide on and align with iOS. See thread [here](https://github.com/open-feature/swift-sdk/pull/46/files#r1894012075)